demos/palm/web/textfx/README.md

```diff
@@ -0,0 +1,110 @@
+
+_TextFX_ uses this same strategy to prime the model to perform the tasks associated with each of its tools. You can find all the prompts in `src/lib/priming.js`, or view them in the app by clicking "Look Under the Hood."
+
+## Requirements
+
+- Node.js (version 18.15.0 or higher)
+- Firebase project
+
+Make sure you have `npm` or `yarn` set up on your machine.
+
+## Developer Setup
+
+Although the PaLM API provides a [REST resource](https://developers.generativeai.google/api/rest/generativelanguage/models?hl=en), it is best practice to avoid embedding API keys directly into code (or in files inside your application's source tree). If you want to call the PaLM API from the client side as we do in this repo, we recommend using a Firebase project with the Call PaLM API Securely extension enabled.
+
+To set up Firebase:
+
+1. Create a Firebase project at https://console.firebase.google.com.
+
+2. Add a web app to your Firebase project and follow the on-screen instructions to add or install the Firebase SDK.
+
+3. Go to https://console.cloud.google.com and select your Firebase project. Then go to _Security > Secret Manger_ using the left-side menu and make sure the Secret Manager API is enabled.
+
+4. If you don't already have an API key for the PaLM API, follow [these instructions](https://developers.generativeai.google/tutorials/setup) to get one.
+
+5. Install the Call PaLM API Securely extension from the [Firebase Extensions Marketplace](https://extensions.dev/extensions). Follow the on-screen instructions to configure the extension.
+
+   **NOTE**: Your project must be on the Blaze (pay as you go) plan to install the extension.
+
+6. Enable anonymous authentication for your Firebase project by returning to https://console.firebase.google.com and selecting _Build_ in the left panel. Then go to _Authentication > Sign-in method_ and make sure _Anonymous_ is enabled.
+
+7. Return to https://console.cloud.google.com and select your Firebase project. Click _More Products_ at the bottom of the left-side menu, then scroll down and click _Cloud Functions_. Select each `ext-palm-secure-backend` Cloud Function and then click _Permissions_ at the top. Add `allUsers` to the Cloud Functions Invoker role.
+
+The above instructions assume that your Firebase backend will be used for individual/experimental purposes. If you anticipate broader usage, consider enabling App Check (see https://firebase.google.com/docs/app-check for an in-depth implementation guide).
+
+To run the application locally:
+
+1. Clone this repo to your local machine.
+
+2. Run `npm i` or `yarn` in the root folder to install dependencies.
```

Comment:
This didn't work for me straight up.  Husky didn't like being installed when the .git dir isn't adjacent: `husky - .git can't be found`.

I got around it with `ln -s ../../../../.git` in the textfx dir.

Reply:
Gotcha—I think this dependency can actually be removed, as I didn't copy over any of the git hooks. Will update in a separate PR.